### ISSUE TITLE
MissingEndParenthesisInMethodCall

### ISSUE THREAD

**Issue by dpetzel** [smurawski/chef-zero-scheduled-task#25]
I originally chimed in with these observations: https://github.com/smurawski/chef-zero-scheduled-task/issues/16#issuecomment-197090440. For the sake of future readers the original symptom was that the provisioner would hang forever at `SUCCESS: Attempted to run the scheduled task "chef-tk".`

However after some debugging I found that in my case the issue seems to be that the scheduled tasks exists immediately and/or without any output to the stream causing things to hang forever. 

The problem (in 0.0.12) appears to be an invalid ps script. If I manually run the command that the tasks is defined with I get the following error:

```
Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation.  All rights reserved.

C:\Users\user>powershell -executionpolicy unrestricted -File C:\Users\User\AppData\Local\Temp\kitchen\chef-client-script.ps1
Missing ')' in method call.
At C:\Users\user\AppData\Local\Temp\kitchen\chef-client-script.ps1:10 cha
r:27
+     $pipewriter.writeline( <<<< SCHEDULED_TASK_DONE: $LastExitCode);
    + CategoryInfo          : ParserError: (CloseParenToken:TokenId) [], Paren
   tContainsErrorRecordException
    + FullyQualifiedErrorId : MissingEndParenthesisInMethodCall
```

I have downgraded to 0.0.10 and things are working again, so it seems to be a regression introduced after 0.0.10.

**Comment by jkerry** [comment]
Replicated on a stock windows 2008r2 image

**Comment by smurawski** [comment]
Yeah, there is a difference in how `$executioncontext.invokecommand.expandstring` works on PowerShell 2 and PowerShell 3/4.

**Comment by smudgerdan** [comment]
@jkerry 
> Replicated on a stock windows 2008r2 image

On my stock 2008R2 image I can't get 0.0.10 working without turning on the .Net 3.5.1 feature.  Is this what you also had to do? Or am I missing something?

**Comment by smurawski** [comment]
No, .net 3.5.1 is required for named pipes.  As soon as [this PR get's merged to test-kitchen](https://github.com/test-kitchen/test-kitchen/pull/1012), I'm going to deprecate this provisioner - it's basically a hack to get around winrm context issues and fixing this in the transport is a much better story (and can be used by verifiers too).

I'll try to get one more release out that fixes the 2008r2 behavior and warns about .net 3.5.1 but it won't be for about a week or so due to some upcoming travel.

**Comment by smudgerdan** [comment]
Ok no worries will just need to add this Windows feature to our base images rather only having chef cookbooks do it.

**Comment by mwrock** [comment]
@smudgerdan if you wanted to try out the Test-Kitchen PR that integrates scheduled task functionality in the core winrm transport, let me know and I can provide instructions. That will likely get you around the issues here and the more success cases reported with that change, the faster it will make it into the next chefdk.

**Comment by smudgerdan** [comment]
@mwrock Thanks, have just seen your reply on [Feature Request: Windows Scheduled Task provisioner #3117](https://github.com/test-kitchen/test-kitchen/pull/1012#issuecomment-216685606) on how to test this new feature and will give it ago tonight, thanks again

**Comment by smurawski** [comment]
I've released v0.2.0 of this gem which deprecates its use and suggests the use of the `elevated` configuration on the transport.
